### PR TITLE
Search functionality for kore

### PR DESCRIPTION
**Pull request by traiansf**
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

File: src/main/haskell/kore/app/exec/Main.hs [hunk 1]
```diff
@@ -304,6 +384,39 @@ mainDefinitionParse = mainParse fromKore
 mainPatternParse :: String -> IO CommonKorePattern
 mainPatternParse = mainParse fromKorePattern
 
+-- | IO action that parses a kore pattern from a filename, verifies it,
+-- converts it to a pure patterm, and prints timing information.
```
Comment (15joeybloom) [inline src/main/haskell/kore/app/exec/Main.hs]: patterm -> pattern

File: src/main/haskell/kore/src/Kore/Step/Condition/Evaluator.hs [hunk 2]
```diff
@@ -84,8 +83,12 @@ evaluate
     (patt, _proof) <- simplifier (unwrapPredicate predicate')
     let
         (subst, _proof) =
-            asPredicateSubstitution (OrOfExpandedPattern.toExpandedPattern patt)
+            give (symbolOrAliasSorts tools)
+            $ asPredicateSubstitution (OrOfExpandedPattern.toExpandedPattern patt)
```
Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/Step/Condition/Evaluator.hs]: Line too long

Comment (virgil-serbanuta) [inline src/main/haskell/kore/src/Kore/Step/Condition/Evaluator.hs]: Line too long
```suggestion
            $ asPredicateSubstitution
                (OrOfExpandedPattern.toExpandedPattern patt)
```

**Comment by ttuegel** [comment]
This was merged in #284.

**Pull request closed without merging**